fluentrequest_test.go

```diff
@@ -2,106 +2,102 @@ package fluentrequest
-	})
-
-	header := http.Header{
-		"Content-Type": {"application/json; charset=UTF-8"},
-	}
-
-	resp, err := FluentRequest().
-		Method(http.MethodPost).
-		Url("https://jsonplaceholder.typicode.com/posts").
-		Body(bytes.NewBuffer(data)).
-		Header(header).
-		Run()
-
-	body, _ := ioutil.ReadAll(resp.Body)
-
-	assert.NoError(t, err)
-	assert.Equal(t, expectedPOSTBody, string(body))
-	assert.Equal(t, http.StatusCreated, resp.StatusCode)
+type ResponseResult struct {
+	responseBody string
+	statusCode   int
 }
 
-func TestPUT(t *testing.T) {
-	data, _ := json.Marshal(map[string]interface{}{
-		"id":     1,
-		"title":  "delectus aut autem",
-		"body":   "bar",
-		"userId": 1,
-	})
-
-	header := http.Header{
-		"Content-Type": {"application/json; charset=UTF-8"},
+func TestRequest(t *testing.T) {
+	tests := []Test{
+		{
+			name:   "Test GET request",
+			method: http.MethodGet,
+			url:    "https://jsonplaceholder.typicode.com/todos/1",
+			want: ResponseResult{
+				statusCode:   http.StatusOK,
+				responseBody: `{ "id": 1, "userId": 1, "title": "delectus aut autem", "completed": false }`,
+			},
+		},
+		{
+			name:   "Test POST request",
+			method: http.MethodPost,
+			url:    "https://jsonplaceholder.typicode.com/todos/",
+			body:   `{ "id": 201, "userId": 2, "title": "foo", "body": "bar", "completed": true }`,
+			want: ResponseResult{
+				statusCode:   http.StatusCreated,
+				responseBody: `{ "id": 201, "userId": 2, "title": "foo", "body": "bar", "completed": true }`,
+			},
+		},
+		{
+			name:   "Test PUT request",
+			method: http.MethodPut,
+			url:    "https://jsonplaceholder.typicode.com/todos/1",
+			body:   `{ "id": 1, "userId": 1, "title": "foo", "body": "bar", "completed": false }`,
+			want: ResponseResult{
+				statusCode:   http.StatusOK,
+				responseBody: `{ "id": 1, "userId": 1, "title": "foo", "body": "bar", "completed": false }`,
+			},
+		},
+		{
+			name:   "Test PATCH request",
+			method: http.MethodPatch,
+			url:    "https://jsonplaceholder.typicode.com/todos/1",
+			body:   `{ "id": 1, "userId": 1, "title": "foobar", "body": "", "completed": true }`,
+			want: ResponseResult{
+				statusCode:   http.StatusOK,
+				responseBody: `{ "id": 1, "userId": 1, "title": "foobar", "body": "", "completed": true }`,
+			},
+		},
+		{
+			name:   "Test DELETE request",
+			method: http.MethodDelete,
+			url:    "https://jsonplaceholder.typicode.com/todos/1",
+			want: ResponseResult{
+				statusCode:   http.StatusOK,
+				responseBody: `{}`,
+			},
+		},
 	}
 
-	resp, err := FluentRequest().
-		Method(http.MethodPut).
-		Url("https://jsonplaceholder.typicode.com/posts/1").
-		Body(bytes.NewBuffer(data)).
-		Header(header).
-		Run()
-
-	body, _ := ioutil.ReadAll(resp.Body)
-
-	assert.NoError(t, err)
-	assert.Equal(t, expectedPUTBody, string(body))
-	assert.Equal(t, http.StatusOK, resp.StatusCode)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			createRequest(t, tt.method, tt.url, tt.body, tt.want)
```

Comment:
Looks kinda strange having a function like this in a package of Chainned function calls. I Would move the test content (lines 85 to 102) to inside the t.Run directly. Also, maybe we will create tests for request witthout all propeties passed, so the function createRequest will not be able to be used.

After this, EXCELLENT work here! i'm very anxious to merge this PR!

Reply:
That's a really good point. Also, I just noticed that I didn't actually add a way to customize the header in the test. I'll add both of the fixes now.